### DISCUSSION TITLE
Si example error

### DISCUSSION THREAD

**Discussion by TiacWei** [nakib/elphbolt#149] [General]
I successfully tested the SiC example, confirming that the installation of elphbolt is working correctly. I then proceeded with the Si example, calculating all necessary inputs as required by myself. 
However, I encountered the following error:

"___________________________________________________________________________
________________________________________________Calculating interactions...
 plasmon energy =                   Infinity
 #q-vecs =          2
 #q-vecs/image <=          1
At line 102 of file ././src/delta.f90
Fortran runtime error: Index '4' of dimension 3 of array 'tetra_evals' above upper bound of 3

Error termination. Backtrace:
At line 102 of file ././src/delta.f90
Fortran runtime error: Index '4' of dimension 3 of array 'tetra_evals' above upper bound of 3
Error termination. Backtrace:
#0  0x42be3b in __delta_MOD_delta_fn_tetra
        at ././src/delta.f90:102
#0  0x42be3b in __delta_MOD_delta_fn_tetra
        at ././src/delta.f90:102
#1  0x4bc5ec in spectral_head_polarizability_3d
        at ././src/screening.f90:231
#2  0x4bc5ec in __screening_module_MOD_calculate_rpa_dielectric_3d_g0_qpath
        at ././src/screening.f90:353
#1  0x4bc5ec in spectral_head_polarizability_3d
        at ././src/screening.f90:231
#2  0x4bc5ec in __screening_module_MOD_calculate_rpa_dielectric_3d_g0_qpath
        at ././src/screening.f90:353"

I think it may be caused by epw. The input for epw.in is as 
--
&inputepw
  prefix      = 'si'
  amass(1)    = 28.086
  asr_typ = 'simple'

  outdir      = 'siepw/'

  elph        = .t.
  kmaps       = .f.
  epbwrite    = .t.
  epbread     = .f.

  etf_mem     = 1
  epwwrite    = .t.
  epwread     = .f.

  nbndsub     = 8

  wannierize  = .t.
  num_iter    = 1000
  dis_win_max = 17.5
  dis_froz_max = 9.5
  proj(1) = 'Si:sp3' 

  wdata(1) = 'conv_tol = 1.0E-12'
  wdata(2) = 'dis_num_iter = 600'
  wdata(3) = 'dis_mix_ratio = 0.5d0'
  wdata(4) = 'dis_conv_tol = 1.0d-11'
  wdata(5) = 'num_print_cycles  = 50'

  elecselfen  = .false.
  phonselfen  = .false.
  a2f         = .false.

  dvscf_dir   = '../phonon_output/save/'
  filqf       = 'highsymkpts.txt'
  filkf       = 'highsymkpts.txt'
   
  nk1         = 12
  nk2         = 12
  nk3         = 12

  nq1         = 6
  nq2         = 6
  nq3         = 6

 /
      16 cartesian
   0.000000000   0.000000000   0.000000000
  -0.166666667   0.166666667  -0.166666667
  -0.333333333   0.333333333  -0.333333333
   0.500000000  -0.500000000   0.500000000
   0.000000000   0.333333333   0.000000000
  -0.166666667   0.500000000  -0.166666667
   0.666666667  -0.333333333   0.666666667
   0.500000000  -0.166666667   0.500000000
   0.333333333   0.000000000   0.333333333
   0.000000000   0.666666667   0.000000000
   0.833333333  -0.166666667   0.833333333
   0.666666667  -0.000000000   0.666666667
   0.000000000  -1.000000000   0.000000000
   0.666666667  -0.333333333   1.000000000
   0.500000000  -0.166666667   0.833333333
  -0.333333333  -1.000000000   0.000000000

I’ve reviewed the input files and compared them with those used in the SiC example, but I haven’t been able to resolve the issue. Could you provide any insights or suggestions on how to fix this?

**Comment by nakib** [comment]
Dear @TiacWei,

Thanks for letting me know. I mistakenly left some uncommented lines in the `app/elphbolt.f90`. I removed them now. Your calculation should now proceed once you pull and build the new version. Let me know if you have further troubles.

Best,
Nakib
